src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -2138,6 +2153,26 @@ void CheckClosed()
         }
 
         #endregion
+
+        #region Misc
+
+        /// <summary>
+        /// Unbinds reader from the connector.
+        /// Should be called before the connector is returned to the pool.
+        /// </summary>
+        internal void UnbindIfNecessary()
```

Comment:
nit: this is only called from one place (right now), so maybe just inline the logic there?